File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/StyleSheet.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model
{
    /// <summary>
    /// Represents the root of a questionnaire style sheet AST.
    /// </summary>
    public class StyleSheet : QLSNode
    {
        public Identifier Id
        {
            get;
            private set;
        }

        public IEnumerable<Page> Pages
        {
            get;
            private set;
        }

        internal StyleSheet(Identifier id, IEnumerable<Page> pages, TextPosition position)
            : base(position)
        {
            Id = id;
            Pages = pages;
        }

        public override T Accept<T>(IStyleSheetVisitor<T> visitor)
        {
            return visitor.VisitStyleSheet(this);
        }
    }
}

File: KevinRene/src/qls/ast/statement/Stylesheet.java
package qls.ast.statement;

import ql.ast.expression.Identifier;
import qls.ast.Statement;
import qls.ast.visitor.StatementVisitor;

public class Stylesheet extends Statement {	
	private final Identifier identifier;
	private final Block pages;

	public Stylesheet(Identifier identifier, Block pages) {
		this.identifier = identifier;
		this.pages = pages;
	}
	
	public Block getPages() {
		return pages;
	}
	
	public Identifier getIdentifier() {
		return identifier;
	}
	
	@Override
	public <T> T accept(StatementVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public String toString() {
		return "Stylesheet";
	}
}